### PR TITLE
Horus/fix #4036

### LINKED ISSUE #4036: Horus hang up in Black Screen and  Blue Screen
If you choose Blue Background Screen in Telemetry, and go back in to the model sites to the mixer Site the Screen is blocking and show unclear colored screen, see pictures.

Also if you scroll through the Model Menus my system block 3 times during preparing a model, this evening. You need a very long Exit on Main switch.

Must be a big bug.

If you choose the Black Background in Telemetry the system hang up in the mixersite. in both problems you have to do a long switch off the main switch to exit.

### PR DESCRIPTION
**Pull request by 3djc**
Remove 'static' from bitmapbuffer that where causing issues when changing theme. I have checked that this does not create extra sdcard access.

@bsongis : please check this one thouroughly, I'm way out of my depth here !

### CODE REVIEW

File: radio/src/gui/480x272/radio_calibration.cpp [hunk 1]
```diff
@@ -39,14 +39,14 @@ enum CalibrationState {
 
 void drawStick(coord_t x, coord_t y, const BitmapBuffer * background, int16_t xval, int16_t yval)
 {
-  static const BitmapBuffer * stick = BitmapBuffer::load(getThemePath("stick_pointer.png"));
+  const BitmapBuffer * stick = BitmapBuffer::load(getThemePath("stick_pointer.png"));
```
Comment (projectkk2glider) [inline radio/src/gui/480x272/radio_calibration.cpp]: This is not good, the file is loaded every time the LCD is refreshed. And delete is also missing - memory leak.

File: radio/src/gui/480x272/model_mixes.cpp [hunk 2]
```diff
@@ -384,7 +384,7 @@ void displayMixStatus(uint8_t channel)
 
 bool menuModelMixAll(event_t event)
 {
-  static const BitmapBuffer * mpx_mode[] = {
+  const BitmapBuffer * mpx_mode[] = {
```
Comment (bsongis) [inline radio/src/gui/480x272/model_mixes.cpp]: Ah yes nice catch

File: radio/src/gui/480x272/radio_calibration.cpp [hunk 3]
```diff
@@ -39,23 +39,21 @@ enum CalibrationState {
 
 void drawStick(coord_t x, coord_t y, const BitmapBuffer * background, int16_t xval, int16_t yval)
 {
-  static const BitmapBuffer * stick = BitmapBuffer::load(getThemePath("stick_pointer.png"));
-  lcd->drawBitmap(x, y, background);
-  lcd->drawBitmap(x + 2 + STICK_PANEL_WIDTH/2 + STICK_PANEL_WIDTH/2 * xval/RESX, y + 2 + STICK_PANEL_WIDTH/2 - STICK_PANEL_WIDTH/2 * yval/RESX, stick);
+  lcd->drawBitmap(x, y, calibStickBackground);
```
Comment (bsongis) [inline radio/src/gui/480x272/radio_calibration.cpp]: Nice one as well

File: radio/src/gui/480x272/radio_calibration.cpp [hunk 4]
```diff
@@ -39,23 +39,21 @@ enum CalibrationState {
...
+  lcd->drawBitmap(x + 2 + STICK_PANEL_WIDTH/2 + STICK_PANEL_WIDTH/2 * xval/RESX, y + 2 + STICK_PANEL_WIDTH/2 - STICK_PANEL_WIDTH/2 * yval/RESX, calibStick);
 }
 
 void drawSticks()
 {
-  static const BitmapBuffer * background = BitmapBuffer::load(getThemePath("stick_background.png"));
```
Comment (bsongis) [inline radio/src/gui/480x272/radio_calibration.cpp]: Yes

File: radio/src/gui/480x272/radio_calibration.cpp [hunk 5]
```diff
@@ -73,8 +71,7 @@ void drawPots()
 
 void drawMouse()
 {
-  static const BitmapBuffer * background = BitmapBuffer::load(getThemePath("trackp_background.png"));
-  drawStick(STICK_LEFT_X, STICKS_Y+100, background, calibratedStick[11], calibratedStick[12]);
```
Comment (bsongis) [inline radio/src/gui/480x272/radio_calibration.cpp]: Yes

File: radio/src/gui/480x272/radio_calibration.cpp [hunk 6]
```diff
@@ -209,9 +206,9 @@ bool menuCommonCalib(event_t event)
       break;
   }
 
-  static const BitmapBuffer * horus = BitmapBuffer::load(getThemePath("horus.bmp"));
-  if (horus) {
-    lcd->drawBitmap((LCD_W-horus->getWidth())/2, LCD_H-20-horus->getHeight(), horus);
+
+  if (calibHorus) {
+    lcd->drawBitmap((LCD_W-calibHorus->getWidth())/2, LCD_H-20-calibHorus->getHeight(), calibHorus);
```
Comment (bsongis) [inline radio/src/gui/480x272/radio_calibration.cpp]: I think the bitmap is already tested in drawBitmap (if I remember well)

**Comment by bsongis** [comment]
Yes your fixes are perfect.
I made those variables static long ago, to avoid to initialize them too often, but now we have themes and they really need to be updated on theme change, otherwise the pointers go nowhere!

**Comment by bsongis** [comment]
I had to resolve some conflicts with the other PR, would you check I did things right?

**Comment by 3djc** [comment]
I just did push a minor correction, but works perfectly :)

**Comment by bsongis** [comment]
👍

**Pull request merged**